app/Services/Gpao/ManufacturingScrapService.php

```
@@ -2,28 +2,37 @@
     /**
      * Declare scrap for a given manufacturing order and item.
      */
-    public function declareScrap(ManufacturingOrder $order, Item $item, float $quantity, string $reason, ?string $notes = null, ?int $reportedById = null): ManufacturingScrap
+    public function declareScrap(ManufacturingOrder $order, Item $item, float $quantity, string $reason, ?string $notes = null, ?int $reportedById = null, ?Warehouse $warehouse = null): ManufacturingScrap
     {
         if ($quantity <= 0) {
             throw new \InvalidArgumentException('Quantity must be strictly positive.');
         }
 
-        return DB::transaction(function () use ($order, $item, $quantity, $reason, $notes, $reportedById) {
+        $warehouse ??= Warehouse::first();
```

PHPStan (2.2.7)

[error] Call to an undefined static method App\Models\Articles\Warehouse::first().


(staticMethod.notFound)